cv_next/app/actions/cvs/fetchCvs.ts

```diff
@@ -0,0 +1,23 @@
+"use server";
+
+import { getCvById } from "@/server/api/cvs";
+
+export const getCvsFromComments = async (
+  comments: CommentModel[]
+): Promise<CvModel[]> => {
+  const uniqueDocuments: Set<CvModel> = new Set(); // Use a Set to store unique documents
+
+  // Iterate through the comments
+  for (const comment of comments) {
```

Comment:
You might be adding only unique documents, but you're making a call for a cv on each comment (at least each parent comment - which is probably the "busiest" level).
Try instead to filter the unique comment ids beforehand (transform cv ids list from comments list -> use set on that).


Also - if this is related to a different PR and is here only because of a merge - please don't merge branches until we approve a PR, I don't know directly how to associate each change and we're doing unnecessary double work. Let's focus on resolving the issues specific to each PR, and then when the PR is approved merge and resolve conflicts (you can either leave a message that you only merged a branch and will merge the PR later, or simply merge the PR right after).

Reply:
This is from profiles that was already merged
Will update according to what you said